### PR TITLE
[gitlab] initial add of the gitlab module

### PR DESCRIPTION
**Pull request by maschmid**

### CODE REVIEW

**Comment by maschmid** [comment]
even though I have reduced the PomModifier to a simple interface, I think we can trim it further and remove all the modifiers altogether.

File: gitlab/src/main/java/cz/xtf/gitlab/GitLabUtil.java [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+import java.util.stream.Stream;
+
+import cz.xtf.core.openshift.OpenShifts;
+import cz.xtf.gitlab.config.GitLabConfig;
+
+public class GitLabUtil {
+
+	private static final String SEPARATOR = "------";
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(GitLabUtil.class);
```
Comment (Crumby) [inline gitlab/src/main/java/cz/xtf/gitlab/GitLabUtil.java]: We should use Lomboks `@Slf4j` tag to use consistent logging approach across the project.

File: gitlab/src/main/java/cz/xtf/gitlab/GitUtil.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+import org.slf4j.LoggerFactory;
+
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.Session;
+
+import cz.xtf.gitlab.config.GitLabConfig;
+
+public final class GitUtil {
+	private static final Logger LOGGER = LoggerFactory.getLogger(GitUtil.class);
```
Comment (Crumby) [inline gitlab/src/main/java/cz/xtf/gitlab/GitUtil.java]: `@Slf4j`

File: gitlab/src/main/java/cz/xtf/gitlab/GitLabUtil.java [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+	public boolean deleteProject(String name) {
+		Stream<GitlabProject> projects = GitLabConfig.group() != null ? getProjects(getGroupId()) : getProjects();
+		return projects.filter(p -> p.getName().equals(name) || p.getName().startsWith(name + SEPARATOR))
+				.map(project -> {
+					try {
+						performDeleteProject(project.getId());
+						FileUtils.deleteDirectory(pathToProject(project.getName()).toFile());
+						LOGGER.debug("Project \"" + name + "\" was deleted");
+					} catch (IOException e) {
```
Comment (Crumby) [inline gitlab/src/main/java/cz/xtf/gitlab/GitLabUtil.java]: Could we polish this catch block a little? Make the commands rather one-liners.

File: gitlab/src/main/java/cz/xtf/gitlab/GitLabUtil.java [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+
+	private void performDeleteProject(Serializable projectId) throws IOException {
+		api.deleteProject(projectId);
+	}
+
+	private Integer getGroupId() {
+		final String group = GitLabConfig.group();
+		if (group != null) {
+			return getGroupId(GitLabConfig.group());
+		}
```
Comment (Crumby) [inline gitlab/src/main/java/cz/xtf/gitlab/GitLabUtil.java]: `else {` -> `} else {`

File: gitlab/src/main/java/cz/xtf/gitlab/GitUtil.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+	public static GitProject cloneRepository(Path target, String url, String branch) {
+		return cloneRepository(target, url, null, null, branch);
+	}
+
+	public static GitProject cloneRepository(Path target, String url, String username, String password) {
+		return cloneRepository(target, url, username, password, "master");
+	}
+
+	public static GitProject cloneRepository(Path target, String url, String username, String password, String branch) {
+
```
Comment (Crumby) [inline gitlab/src/main/java/cz/xtf/gitlab/GitUtil.java]: Remove empty line.

File: gitlab/src/main/java/cz/xtf/gitlab/GitLabUtil.java [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+					project = api.createProject(name, groupId, null, null, null, null, null, null, true,
+							null, url);
+				} catch (IOException x) {
+					lastExcepion = x;
+					if (!x.getMessage().contains(
+							WAIT_FOR_GITLAB_CREATE_PROJECT_MESSAGE)
+							|| attempt >= WAIT_FOR_GITLAB_CREATE_PROJECT_MAX_ATTEMPTS) {
+						throw x;
+					} else {
+						LOGGER.debug(
```
Comment (Crumby) [inline gitlab/src/main/java/cz/xtf/gitlab/GitLabUtil.java]: Could we make this one liner using standard logger syntax? log.debug("Some code with env value: {}", envValue);

File: gitlab/src/main/java/cz/xtf/gitlab/GitLabUtil.java [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+		api.deleteProject(projectId);
+	}
+
+	private Integer getGroupId() {
+		final String group = GitLabConfig.group();
+		if (group != null) {
+			return getGroupId(GitLabConfig.group());
+		}
+		else {
+			return getGroupId(OpenShifts.master().getNamespace());
```
Comment (Crumby) [inline gitlab/src/main/java/cz/xtf/gitlab/GitLabUtil.java]: `XtfConfig.namespace()`?

**Pull request closed without merging**